x-pack/plugins/enterprise_search/server/index.ts

```diff
@@ -14,6 +14,8 @@ export const plugin = (initializerContext: PluginInitializerContext) => {
 
 export const configSchema = schema.object({
   host: schema.maybe(schema.string()),
+  enabled: schema.boolean({ defaultValue: true }),
+  accessCheckTimeout: schema.number({ defaultValue: 600 }),
```

Comment:
This was at 200 in the original proof-of-concept, but that's definitely not enough from my local testing. I was getting timeout errors at 400ms and below in any condition. 500 passed over 3 times for users with access, and 600 was the requirement for users without access (I assume the redirect from AS/WS takes extra time).

My hesitation here is that all these are with localhost values, which doesn't account for real-world production/distance latency. What are people's thoughts on increasing this to 1000 to give ourselves a buffer?

Reply:
Resolved via email thread - timeout to 5000 and add a shorter 300 warning timeout